io.sloeber.ui/src/io/sloeber/ui/monitor/internal/SerialListener.java

```diff
@@ -122,16 +125,49 @@ public void dispose() {
 	public class TxtUpdater implements Runnable {
 		private boolean running = false;
 		private String additionalSerialData = new String();
+		private StringBuilder lineBuffer = new StringBuilder();
+
+		private DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HH:mm:ss"); //$NON-NLS-1$
+		private static final String arduinoEOL = "\r\n"; //$NON-NLS-1$
```

Comment:
It looks to me as if you are assuming arduino always sends \r\n as newline. This is however not true.
\n is a safer option.

Reply:
I did wonder about this.

How about I discard all \r and match on \n. A simple implementation of that might be slow but I'll see how it goes.

Reply:
I quite often use String.replace ("\r\n","n");
However in this case I don't really like it because you can save the monitor content for later processing and then the user may not get what he wants.